src/formula.rs

```diff
@@ -118,14 +102,10 @@ impl Formula {
     /// and returns the range the newly added string occupies on the buffer.
     ///
     /// See [`crate::verify`] for more about this format.
-    pub fn append_to_stack_buffer(
-        &self,
-        stack_buffer: &mut Vec<u8>,
-        sset: &Arc<SegmentSet>,
-        nset: &Arc<Nameset>,
-    ) -> Range<usize> {
+    fn append_to_stack_buffer(&self, stack_buffer: &mut Vec<u8>, db: &Database) -> Range<usize> {
         let tos = stack_buffer.len();
-        for symbol in self.iter(sset, nset) {
+        let nset = &**db.name_result();
+        for symbol in self.as_ref(db) {
             fast_extend(stack_buffer, nset.atom_name(symbol));
             *stack_buffer.last_mut().unwrap() |= 0x80;
         }
```

Comment:
This this needs the database, maybe this could rather be a function of `FormulaRef` ?